lib/jobs/ucs-service-profile-discovery.js

```diff
@@ -0,0 +1,197 @@
+        return this.ucs.clientRequest(url)
+            .then(function(response) {
+                return response.body;
+            });
+    };
+
+
+    /**
+     * @function getRoot
+     * @description return all the elements from the Cisco UCS Manager that RackHD cares about
+     */
+    UcsServiceProfileDiscoveryJob.prototype.getRoot = function () {
+        var username = this.settings.username;
+        var password = this.settings.password;
+        var ucs = this.settings.ucs;
+        var url=
+            "/serviceProfile?host=" + ucs+ "&user="+ username + "&password=" + password;
+        return this.ucs.clientRequest(url)
+            .then(function(response) {
+                return response.body;
+            });
+    };
+
+    /**
+     * @function createServiceProfile
+     * @description discovers all the service profiles in the Cisco UCS
+     */
+    UcsServiceProfileDiscoveryJob.prototype.createServiceProfile = function (root) {
+        var self = this;
+        var username = this.settings.username;
+        var password = this.settings.password;
+        var ucs = this.settings.ucs;
+        var baseurl = this.settings.protocol+"://"+this.settings.host+ ":"+this.settings.port;
+        var url=
+            baseurl + "/serviceProfile?host=" + ucs+ "&user="+ username + "&password=" + password;
+        return Promise.try(function () {
+            if (_.has(root, 'ServiceProfile')) {
+                return self.ucs.clientRequest(url)
+                    .then(function(res) {
+                        assert.object(res);
+                        return res.body;
+                    })
+                    .then(function(data) {
+                        assert.object(data);
+                        return data['ServiceProfile']['members']
+                    })
+                    .map(function(members){
+                        self.createNode(members, Constants.NodeTypes.Compute);
+                    });
+            }else{
+                logger.warning('No Logical Servers found');
+                return;
+            }
+        });
+    };
+
+    UcsServiceProfileDiscoveryJob.prototype.createNode = function(data, type) {
+
+        var self = this;
+        var config = Object.assign({}, self.settings);
+
+        assert.string(data.name);
+        assert.string(data.path);
+        var id;
+        id = [config.ucs, data.path];
+        var obm = {
+            config: config,
+            service: 'ucs-obm-service'
+        };
+        var node = {
+            type: type,
+            name: data.name,
+            associatedServer: data.associatedServer,
+            identifiers: id,
+            relations : []
+        };
+        // Update existing node  or create a new one
+        return waterline.nodes.needOne({identifiers:data.path})
+            .then(function(curNode) {
+                return waterline.nodes.updateOne(
+                    { id: curNode.id },
+                    node
```

Comment:
In line 175, the relations is reset to be empty, since this line of code is to update the node document, so whether it is intentionally to reset the relations? And where the relations are built again?